### ISSUE TITLE
Feature request: Custom deserialization of scalars/object types/interface types

### ISSUE THREAD

**Issue by capaj** [helios1138/graphql-typed-client#2]
it would be nice to be able to associate scalars/object types/interface types to classes at runtime.

So currently the whole client is build statically so it's impossible, but ideally I could do something like:

```ts
class User {
  get fullName () {
     return this.firstName + ' ' + this.lastName
  }
}

const myCustomClient = graphqlTypedClient({DateTime: Moment, User: User})

const {data} = myCustomClient.query({currentUser: {firstName: 1, lastName: 1}})
// if this returns {data: {currentUser: {firstName: 'Frodo', lastName: 'Baggins'}}}
console.log(data.currentUser.fullName) // would give me the user's full name-so 'Frodo Baggins'

```

also being able to define a `function` as custom deserializer would be beneficial. So not just a class.

**Comment by capaj** [comment]
another thought-it would be nice to be able to turn it off/override per query. There might be cases when you query data you might not want to deserialize.
So ideally I could pass the same hashmap as another argument to query/mutation/subscription. If I pass a falsy value it should just give me raw JSON values even if I have top level deserialization map.

**Comment by helios1138** [comment]
> would be nice to be able to turn it off/override per query

@capaj can you describe the case where you would want both deserialized and original values (or same types deserialized differently) in the same application?

Rendering custom types based on some deserialization mapping provided during client generation seems easy enough, but I don't see an easy way to do it dynamically during execution

**Comment by helios1138** [comment]
@capaj custom scalar type mapping implemented in v1.2.0. Object mapping is more tricky and probably lies outside of the scope of this project.
